code/addons/test/src/vitest-plugin/index.ts

```diff
@@ -144,79 +226,22 @@ export const storybookTest = (options?: UserOptions): Plugin => {
-        __VITEST_EXCLUDE_TAGS__: finalOptions.tags.exclude.join(','),
-        __VITEST_SKIP_TAGS__: finalOptions.tags.skip.join(','),
-      };
-
-      config.envPrefix = Array.from(new Set([...(config.envPrefix || []), 'STORYBOOK_', 'VITE_']));
-
-      if (config.test.browser) {
-        config.test.browser.screenshotFailures ??= false;
-      }
-
-      // copying straight from https://github.com/vitejs/vite/blob/main/packages/vite/src/node/constants.ts#L60
-      // to avoid having to maintain Vite as a dependency just for this
-      const viteDefaultClientConditions = ['module', 'browser', 'development|production'];
-
-      config.resolve ??= {};
-      config.resolve.conditions ??= [];
-      config.resolve.conditions.push(
-        'storybook',
-        'stories',
-        'test',
-        ...viteDefaultClientConditions
-      );
-
-      config.test.setupFiles ??= [];
-      if (typeof config.test.setupFiles === 'string') {
-        config.test.setupFiles = [config.test.setupFiles];
-      }
-      config.test.setupFiles.push('@storybook/experimental-addon-test/internal/setup-file');
+      // return the new config, it will be deep-merged by vite
+      return config;
+    },
+    async configureServer(server) {
+      const statics: ReturnType<typeof mapStaticDir>[] = [];
+      const staticDirs = await presets.apply('staticDirs', []);
 
-      // when a Storybook script is provided, we spawn Storybook for the user when in watch mode
-      if (finalOptions.storybookScript) {
-        config.test.globalSetup = config.test.globalSetup ?? [];
-        if (typeof config.test.globalSetup === 'string') {
-          config.test.globalSetup = [config.test.globalSetup];
+      for (const staticDir of staticDirs) {
+        try {
+          statics.push(mapStaticDir(staticDir, directories.configDir));
+        } catch (e) {
+          console.warn(e);
         }
-        config.test.globalSetup.push('@storybook/experimental-addon-test/internal/global-setup');
       }
 
-      config.test.server ??= {};
-      config.test.server.deps ??= {};
-      config.test.server.deps.inline ??= [];
-      if (Array.isArray(config.test.server.deps.inline)) {
-        config.test.server.deps.inline.push('@storybook/experimental-addon-test');
-      }
-
-      config.optimizeDeps ??= {};
-      config.optimizeDeps = {
-        ...config.optimizeDeps,
-        include: [...(config.optimizeDeps.include ?? []), '@storybook/experimental-addon-test/**'],
-      };
-
-      if (frameworkName?.includes('react') || frameworkName?.includes('nextjs')) {
-        config.optimizeDeps.include.push('react-dom/test-utils');
-      }
-
-      if (frameworkName?.includes('vue3')) {
-        config.define ??= {};
-        config.define.__VUE_PROD_HYDRATION_MISMATCH_DETAILS__ = 'false';
-      }
-    },
-    configureServer(server) {
-      statics.map(({ staticPath, targetEndpoint }) => {
+      statics.forEach(({ staticPath, targetEndpoint }) => {
```

Comment:
In his defence, it has actually just been moved. https://github.com/storybookjs/storybook/pull/29806/files#diff-604740f662e4e19338c6063aef059bbac9f56dee9c0f378db77d98ae702c2709L118-L126

Not sure why though. It would make sense to move `const staticDirs = await presets.apply('staticDirs', []);` up into the `Promise.all` list.